client/components/chart/index.jsx

```diff
@@ -137,11 +137,28 @@ function Chart( {
 
 	const { isTooltipVisible, tooltipContext, tooltipPosition, tooltipData } = tooltip;
 
+	const ChartYAxis = () => (
+		<div ref={ yAxisRef } className="chart__y-axis">
+			<div className="chart__y-axis-width-fix">{ numberFormat( 1e5 ) }</div>
+			<div className="chart__y-axis-label is-hundred">
+				{ yMax > 1 ? numberFormat( yMax ) : numberFormat( yMax, 2 ) }
+			</div>
+			<div className="chart__y-axis-label is-fifty">
+				{ yMax > 1 ? numberFormat( yMax / 2 ) : numberFormat( yMax / 2, 2 ) }
+			</div>
+			<div className="chart__y-axis-label is-zero">{ numberFormat( 0 ) }</div>
+		</div>
+	);
+
 	// This is a hack to avoid the flickering on page load.
 	// The component listens on the resize event of its own, which would resize on initialization.
 	// The hack renders an empty div, which triggers the resize event, and then the actual component would be rendered.
-	if ( width <= 0 ) {
-		return <div ref={ resizeRef }></div>;
+	if ( sizing.clientWidth <= 0 || yAxisSize.clientWidth <= 0 ) {
```

Comment:
What's the difference between `width`, `sizing.clientWidth`, and `yAxisSize.clientWidth`?

Reply:
@jsnmoon Good question.

- `width` is the calculated width to render the chart
- `sizing.clientWidth` is from the resize callback after the rendering of the chart
- `yAxisSize.clientWidth?` is the the resize callback after the rendering of the Y axis

The hack basically pre-render a div to get the widths so that the component could consitently be rendered rather than rendering itself with initial width (char: 650px, y axis: 0) and then re-rendering several times.

Reply:
Got it; thanks for the explanation!